### PR TITLE
Fix: sanitize donor wall query args

### PR DESCRIPTION
**Pull request by jonwaldstein**
## Description

It came to our attention that some shortcode request args for the donor wall were not sanitized properly.  This PR ensures proper sanitization of those args.

## Affects

The `[give_donor_wall]` shortcode and Block

## Visuals

N/A

## Testing Instructions

Make sure the `[give_donor_wall]` short-code and Donor Wall Block still work as expected

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203721154588421

### CODE REVIEW

File: includes/donors/class-give-donor-wall.php [hunk 1]
```diff
@@ -356,8 +356,8 @@ private function get_query_param( $atts = [] ) {
 		$query_atts['orderby']       = in_array( $atts['orderby'], $valid_orderby ) ? $atts['orderby'] : 'post_date';
 		$query_atts['limit']         = $atts['donors_per_page'];
 		$query_atts['offset']        = $atts['donors_per_page'] * ( $atts['paged'] - 1 );
-		$query_atts['form_id']       = implode( '\',\'', explode( ',', $atts['form_id'] ) );
-		$query_atts['ids']           = implode( '\',\'', explode( ',', $atts['ids'] ) );
+        $query_atts['form_id']       = $this->split_string(implode('\',\'', explode(',', $atts['form_id'])), 'absint');
```
Comment (kjohnson) [inline includes/donors/class-give-donor-wall.php:359]: Locally this is producing an error in that it breaks the query. This line should be resulting in something like `1','2','3`, which is used directly in the query.

Using `$this->split_string` here results in an `array`.

Comment (kjohnson) [reply includes/donors/class-give-donor-wall.php:359]: Instead we can just introduce `array_map('absint', ...)` after we `explode` and before we `implode`.

```
implode( '\',\'', array_map( 'absint', explode( ',', $atts['form_id'] ) ) );
```

Comment (kjohnson) [reply includes/donors/class-give-donor-wall.php:359]:
```suggestion
        $query_atts['form_id'] = implode( '\',\'', array_map( 'absint', explode( ',', $atts['form_id'] ) ) );
```

File: includes/donors/class-give-donor-wall.php [hunk 2]
```diff
@@ -356,8 +356,8 @@ private function get_query_param( $atts = [] ) {
...
+        $query_atts['ids']           = $this->split_string(implode('\',\'', explode(',', $atts['ids'])), 'absint');
```
Comment (kjohnson) [inline includes/donors/class-give-donor-wall.php:360]:
```suggestion
        $query_atts['ids']           = implode( '\',\'', array_map( 'absint', explode( ',', $atts['ids'] ) ) );
```

**Review by kjohnson — APPROVED** [review]
Can confirm that this resolves the issue.

**Review by Benunc — APPROVED** [review]
this passes all automated tests, and I also manually added both a block and a shortcode to a page.

**Pull request merged**